src/app/beer_garden/db/schemas/garden_schema.py

```diff
@@ -0,0 +1,133 @@
+import logging
+
+from brewtils.models import Garden as BrewtilsGarden
+from brewtils.schemas import StatusInfoSchema  # noqa # until we can fully decouple
+from brewtils.schemas import SystemSchema  # noqa # until we can fully decouple
+from marshmallow import Schema, ValidationError, fields
+from marshmallow.decorators import post_load, pre_dump, pre_load, validates_schema
+from mongoengine.queryset.queryset import QuerySet
+
+logger = logging.getLogger(__name__)
+
+
+class GardenBaseSchema(Schema):
+    """Class to give Marshmallow Schemas the desired behavior of throwing
+    exceptions on errors when marshalling/unmarshalling. Otherwise, each line of code
+    utilizing these would need to pull apart MarshalResult objects in order to return
+    a meaningful error."""
+
+    @pre_load
+    @pre_dump
+    @validates_schema(skip_on_field_errors=False)
+    def validate_all_keys(self, data, **kwargs):
+        # do not allow extraneous keys if working on a dictionary
+        if isinstance(data, dict):
+            extra_args = [key for key in data.keys() if key not in self.fields]
+
+            if len(extra_args) > 0:
+                raise ValidationError(
+                    f"Only {', '.join(self.fields.keys())} allowed as keys; "
+                    f"these are not allowed: {', '.join(extra_args)}"
+                )
+
+    def validate(self, data, many=None, partial=None):
+        if isinstance(data, dict):
+            errors = super().validate(data, many=many, partial=partial)
+            schema_key = "_schema"
+
+            if len(errors) > 0:
+                if schema_key in errors:
+                    raise ValidationError(errors[schema_key].pop())
+                error_str = ""
+                for key in errors:
+                    value = errors[key]
+                    if isinstance(value, list):
+                        update_value = value.pop()
+                    else:
+                        update_value = str(value)
+                    error_str += key + ": " + update_value
+                raise ValidationError(error_str)
+
+            return {}
+
+    def load(self, data, many=None, partial=None):
+        _ = self.validate(data, many=many, partial=partial)
+        return super().load(data, many=many, partial=partial)
+
+    def dump(self, obj, many=None, update_fields=True, **kwargs):
+        if isinstance(obj, (list, QuerySet)):
+            many = True
+
+        _ = self.validate(obj, many=many)
+        dumped = super().dump(obj, many=many, update_fields=update_fields, **kwargs)
+
+        return dumped
+
+
+def _port_validator(value):
+    return 1024 <= value < 65535
```

Comment:
I don't think anything would preclude a garden from having it's public connection on a port lower than 1024.  That part of the check seems unnecessarily limiting.